script.js

```diff
@@ -1,4 +1,346 @@
+var mode = "numberOfPlayers";
+var noOfPlayers = 0;
+var noOfDice = 0;
+var leaderboardArray = [];
+var diceRollsArray = [];
+var currentRoundCombinedNums = [];
+var autoMode = false;
+var knockoutMode = false;
+var isNormalMode = false;
+var currentPlayer = 0;
 var main = function (input) {
-  var myOutputValue = 'hello world';
+  if (mode == "numberOfPlayers") {
+    var noOfPlayersOutput = setPlayerNumber(input);
+    return noOfPlayersOutput;
+  }
+  if (mode == "numberOfDice") {
+    var noOfDiceOutput = setDiceNumber(input);
+    return noOfDiceOutput;
+  }
+  if (mode == "autoModeSelect") {
+    var autoGenerateMode = setAutoMode(input);
+    return autoGenerateMode;
+  }
+  if (mode == "knockoutModeSelect") {
+    var knockoutModeOutput = setKnockoutMode(input);
+    return knockoutModeOutput;
+  }
+  if (mode == "gameModeSelect") {
+    var gameModeOutput = setGameMode(input);
+    return gameModeOutput;
+  } else {
+    var diceRolls = multipleDiceRoll(noOfPlayers, noOfDice, input);
+    return diceRolls;
+  }
```

Comment:
good job setting the logic for each game mode into individual functions. It would be great if you could place the individual helper functions outside of the main function too!